### PR TITLE
F UI/service discovery

### PR DESCRIPTION
**Pull request by philrenaud**
UI for Nomad Service Discovery. Resolves https://github.com/hashicorp/nomad/issues/14193

Adds:
- Health checks to the services table on allocation pages
- A new Services route within a job
- Service details fly-out from allocation pages
and more!

image
image

### CODE REVIEW

File: ui/mirage/models/job.js [hunk 1]
```diff
@@ -4,4 +4,5 @@ export default Model.extend({
   taskGroups: hasMany('task-group'),
   jobSummary: belongsTo('job-summary'),
   jobScale: belongsTo('job-scale'),
+  // services: hasMany('service'),
```
Comment (ChaiWithJai) [inline ui/mirage/models/job.js:7]: `comment` `non-blocking`: Do you want to commit this?

File: ui/tests/acceptance/job-services-test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { module, test } from 'qunit';
+import { find, findAll, currentURL, settled } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { allScenarios } from '../../mirage/scenarios/default';
+
+import a11yAudit from 'nomad-ui/tests/helpers/a11y-audit';
+import Services from 'nomad-ui/tests/pages/jobs/job/services';
+
+module('Acceptance | job services', function (hooks) {
```
Comment (ChaiWithJai) [inline ui/tests/acceptance/job-services-test.js:10]: `question`: Have you run this test a few times? I've noticed that this has been flaky.

**Review by ChaiWithJai — APPROVED** [review]
LGTM. Made a few minor comments -- one pointing out a flaky test.

**Review by shoenig — COMMENTED** [review]
Nice!

File: ui/tests/acceptance/job-services-test.js (hunk 2 above)
Comment (philrenaud) [reply ui/tests/acceptance/job-services-test.js:10]: I've only encountered flakiness when running other tests against its test scenario; I'll keep monitoring it and isolate the scenario within the test itself if it keeps flaking out.

**Pull request merged**